doc/man3/OBJ_nid2obj.pod

```diff
@@ -155,6 +155,8 @@ a NID or B<NID_undef> on error.
 
 OBJ_add_sigid() returns 1 on success or 0 on error.
 
+OBJ_obj2txt() returns the number of bytes written to I<buf> or -1 on error. 
```

Comment:
I think that is inaccurate. As far as I know it returns the size that would be written if the buffer was big enough. I'm not sure if we properly document how we truncate it in case it's not big enough.

Reply:
totally agree with you. How about **OBJ_obj2txt() returns the number of bytes written to I<buf> if I<buf_len> is big enough. Otherwise, the result is truncated and the total amount of space required is returned. It also returns -1 on error.**